app/models/talentlms/course.rb

```diff
@@ -68,5 +85,31 @@ def completed_by?(user)
 
       Talentlms::CompletedTraining.where(login_id: login.id, course_id: id).exists?
     end
+
+    def active_date_order_value
+      [
+        active? ? 0 : 1,
+        start_date ? 0 : 1,
+        start_date,
+        end_date ? 0 : 1,
+        end_date,
+      ]
+    end
```

Comment:
This seems overly complex.  What about a scope, something like:
``
scope :ordered, -> do
 order(active: :desc, Arel.sql(arel_table[:start_date].desc.to_sql + ' NULLS LAST'), Arel.sql(arel_table[:end_date].desc.to_sql + ' NULLS LAST'))
end
```

That's probably non-functional, and there's probably a better way to indicate nulls last, but then the call laster can just be `config.courses.ordered`.

That said, if it's not easy, and this seems to work, we need to instantiate all of the courses anyway, so it's not a big deal.